javagen/src/main/java/com/azure/autorest/template/ServiceSyncClientTemplate.java

```diff
@@ -129,25 +130,21 @@ protected void writeClass(AsyncSyncClient syncClient, JavaClass classBlock, Java
   protected void writeMethods(AsyncSyncClient syncClient, JavaClass classBlock) {
     final ServiceClient serviceClient = syncClient.getServiceClient();
     final MethodGroupClient methodGroupClient = syncClient.getMethodGroupClient();
-    final boolean wrapServiceClient = methodGroupClient == null;
-    if (wrapServiceClient) {
-      serviceClient.getClientMethods().stream()
-          .filter(clientMethod -> clientMethod.getMethodVisibility() == JavaVisibility.Public)
-          .filter(clientMethod -> !clientMethod.isImplementationOnly())
-          .filter(clientMethod -> !clientMethod.getType().name().contains("Async"))
-          .forEach(clientMethod -> {
-            writeMethod(clientMethod, classBlock);
-          });
-    } else {
-      methodGroupClient.getClientMethods().stream()
-          .filter(clientMethod -> clientMethod.getMethodVisibility() == JavaVisibility.Public)
-          .filter(clientMethod -> !clientMethod.isImplementationOnly())
-          .filter(clientMethod -> !clientMethod.getType().name().contains("Async"))
-          .forEach(clientMethod -> {
-            writeMethod(clientMethod, classBlock);
-          });
+
+    List<ClientMethod> clientMethods = serviceClient.getClientMethods();
+    if (CoreUtils.isNullOrEmpty(clientMethods)) {
+      clientMethods = methodGroupClient.getClientMethods();
     }
```

Comment:
This code will not work in MethodGroupClient, if ServiceClient does contain client methods (it always do, whether the wrapper is on ServiceClient or MethodGroupClient)

Reply:
That seems like a bug. Client methods should either be in ServiceClient or MethodGroupClient and not both (even based on the previous code path, either the service client's methods were used or method group client's methods were used). Anyway, I'll use the previous check to switch but something to fix later.